### PR TITLE
Add QA IT to the Unified CI

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

With #19757 we added an new QA module containing integration tests on the root level. This PR should add the execution of such tests to the unified CI. Furthermore, it adds the integrations tests from the Zeebe CI already also to the job, which can be seen as a blue print to add more such integration tests.

Probably useful for Operate, Tasklist and other components that have separate component related integration tests. 

## Related issues

related to #19757

### CODE REVIEW

File: .github/actions/paths-filter/action.yml [hunk 1]
```diff
@@ -11,8 +11,8 @@ outputs:
         github.event_name == 'push' ||
         steps.filter-common.outputs.github-actions-change == 'true'
       }}
-  java-unit-tests:
-    description: Output whether `unit-tests` (java) should be run based on GitHub event and files changed
+  java-code-changes:
+    description: Output whether `java code changes` have been detected
```
Comment (cmur2) [inline .github/actions/paths-filter/action.yml:15]: the reason why I included this "should be run based on GitHub event and files changed" blurb is that this condition also always is true for `push` events and people could assume that is wrong - it is probably a nitpick but just to provide context

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -116,6 +116,99 @@ jobs:
...
+        if: ${{ failure() || cancelled() }}
+        with:
+          name: "[IT] ${{ matrix.name }}"
+      - name: Observe build status
+        if: always()
+        continue-on-error: true
+        uses: ./.github/actions/observe-build-status
+        with:
+          job_name: "integration-tests/${{ matrix.group }}"
+          build_status: ${{ contains(steps.*.conclusion, 'failure') && 'failure' || 'success' }}
```
Comment (cmur2) [inline .github/workflows/ci.yml:207]: `build_status: ${{ job.status }}` again :)

File: .github/actions/paths-filter/action.yml (hunk 1 above)
Comment (Zelldon) [reply .github/actions/paths-filter/action.yml:15]: Not sure what you mean with the comment :) Should I add this back again?

Comment (cmur2) [reply .github/actions/paths-filter/action.yml:15]: I'm not sure myself either what I mean...

I think I'd prefer "Output whether jobs depending on Java code changes should be run based on GitHub event and files changed" here :)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Zelldon) [reply .github/workflows/ci.yml]: Sorry, I could swear I replaced that.

File: .github/actions/paths-filter/action.yml (hunk 1 above)
Comment (Zelldon) [reply .github/actions/paths-filter/action.yml:15]: How about: 

> Output whether jobs depending on Java code changes should be run, related checks are based on GitHub events and java files changes

Comment (cmur2) [reply .github/actions/paths-filter/action.yml]: Fine by me :)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -92,7 +92,7 @@ jobs:
           ./mvnw -T2 -B --no-snapshot-updates
           -D skipITs -D skipQaBuild=true -D skipChecks -D surefire.rerunFailingTestsCount=3
           -D junitThreadCount=16
-          -P skip-random-tests,parallel-tests,extract-flaky-tests
+          -P skip-random-tests,parallel-tests,extract-flaky-tests,skipFrontendBuild
```
Comment (Zelldon) [inline .github/workflows/ci.yml:95]: ℹ️  Removes around 1 min build time

**Review by cmur2 — APPROVED** [review]
Nice!

**Pull request merged**